docs/misc/resource-routes.md

```diff
@@ -0,0 +1,212 @@
+  params,
+}: LoaderFunctionArgs) {
+  return json(await getReport(params.id));
+}
+
+export default function Report() {
+  const report = useLoaderData<typeof loader>();
+  return (
+    <div>
+      <h1>{report.name}</h1>
+      <Link to="pdf" reloadDocument>
+        View as PDF
+      </Link>
+      {/* ... */}
+    </div>
+  );
+}
+```
+
+It's linking to a PDF version of the page. To make this work we can create a Resource Route below it. Notice that it has no default export component: that makes it a Resource Route.
+
+```tsx filename=app/routes/reports.$id[.pdf].tsx
+export async function loader({
+  params,
+}: LoaderFunctionArgs) {
+  const report = await getReport(params.id);
+  const pdf = await generateReportPDF(report);
+  return new Response(pdf, {
+    status: 200,
+    headers: {
+      "Content-Type": "application/pdf",
+    },
+  });
+}
+```
+
+<docs-info>Note the filename uses [escaping patterns][escaping] to include the extension in the URL.</docs-info>
+
+When the user clicks the link from the UI route, they will navigate to the PDF.
+
+## Linking to Resource Routes
+
+<docs-error>It’s imperative that you use <code>reloadDocument</code> on any Links to Resource Routes</docs-error>
+
+There's a subtle detail to be aware of when linking to resource routes. You need to link to it with `<Link reloadDocument>` or a plain `<a href>`. If you link to it with a normal `<Link to="pdf">` without `reloadDocument`, then the resource route will be treated as a UI route. React Router will try to get the data with `fetch` and render the component. Don't sweat it too much, you'll get a helpful error message if you make this mistake.
+
+## Handling different request methods
+
+To handle `GET` requests export a loader function:
+
+```tsx
+import type * as Route from "./+types.resource";
+
+export const loader = async ({
+  request,
+}: Route.LoaderArgs) => {
+  // handle "GET" request
+
+  return {success: true}
+};
+```
+
+To handle `POST`, `PUT`, `PATCH` or `DELETE` requests export an action function:
+
+```tsx
+import type * as Route from "./+types.resource";
+
+export const action = async ({
+  request,
+}: Route.ActionArgs) => {
+  switch (request.method) {
+    case "POST": {
+      /* handle "POST" */
+    }
+    case "PUT": {
+      /* handle "PUT" */
+    }
+    case "PATCH": {
+      /* handle "PATCH" */
+    }
+    case "DELETE": {
+      /* handle "DELETE" */
+    }
+  }
+};
+```
+
+Resource Routes do not support [non-standard HTTP methods][nonstandard-http-methods] - those should be handled by the HTTP server that serves your React Router requests.
+
+## Turbo Stream
+
+Returning bare objects from a Resource Route will automatically encode the responses as a [turbo-stream][turbo-stream], which automatically serializes Dates, Promises, and other objects over the network. React Router automatically deserializes turbo-stream data when you call resource routes from React Router APIs such as `<Form>`, `useSubmit`, or `useFetcher`.
```

Comment:
Might be worth mentioning that `data` does the same thing in case people don't realize.

```suggestion
Returning bare objects or using the [`data` util](../../api/react-router/data) from a Resource Route will automatically encode the responses as a [turbo-stream][turbo-stream], which automatically serializes Dates, Promises, and other objects over the network. React Router automatically deserializes turbo-stream data when you call resource routes from React Router APIs such as `<Form>`, `useSubmit`, or `useFetcher`.
```